### PR TITLE
Add optional rapid sync timestamp field to NetworkGraph

### PR DESCRIPTION
**Pull request by arik-so**
This field will be necessary for rapid gossip sync to not burden users with a one-off variable requiring separate storage.

### CODE REVIEW

File: lightning/src/routing/network_graph.rs [hunk 1]
```diff
@@ -127,6 +127,7 @@ pub struct NetworkGraph {
 	// Lock order: channels -> nodes
 	channels: RwLock<BTreeMap<u64, ChannelInfo>>,
 	nodes: RwLock<BTreeMap<NodeId, NodeInfo>>,
+	last_rapid_gossip_sync_timestamp: Option<u32>,
```
[the remark this answers is not in the document]
Comment (arik-so) [reply lightning/src/routing/network_graph.rs:130]: Yes to 1) and 2), but 3) should be done in a separate PR, preferably the one with the background processor pruning.

[the remark this answers is not in the document]
Comment (arik-so) [reply lightning/src/routing/network_graph.rs]: Lol I thought we wanted smaller, easy-to-read PRs :P

[the remark this answers is not in the document]
Comment (arik-so) [reply lightning/src/routing/network_graph.rs]: with a clear separation of responsibilities :P

[the remark this answers is not in the document]
Comment (arik-so) [reply lightning/src/routing/network_graph.rs]: will do!

**Comment by arik-so** [comment]
I hear you, but it's either that or massive PRs.

File: lightning/src/routing/network_graph.rs [hunk 2]
```diff
@@ -1014,12 +1020,18 @@ impl Readable for NetworkGraph {
 			let node_info = Readable::read(reader)?;
 			nodes.insert(node_id, node_info);
 		}
-		read_tlv_fields!(reader, {});
+
+		let mut last_rapid_gossip_sync_timestamp: Option<u32> = None;
+
```
Comment (valentinewallace) [inline lightning/src/routing/network_graph.rs:1025]: nit: no newline here

**Review by valentinewallace — COMMENTED** [review]
Could we explain this change a little more in the commit message? 

Other than that plus these nits, LGTM

**Comment by arik-so** [comment]
Updated commit message; lmk if I should rename the PR, too. @valentinewallace

**Pull request merged**